testing/it_sidecar/client/sidecar_client.go

```diff
@@ -34,6 +34,20 @@ var setupCMD = flag.String("setup", "", "the path to the it setup command")
 // ready, and then forwards service logs to test output.  On completion, signals to the it_sidecar
 // to teardown the test namespace
 func (s *K8STestSetup) TestMain(m *testing.M) {
+	s.testMain(m, nil)
+}
+
+// TestMainWithHook will execute the provided setup command, wait for configured pods and services to be
+// ready, forwards service logs to test output, and then runs the provided hook for any custom pre-test suite setup.
+// On completion, signals to the it_sidecar to teardown the test namespace.
+func (s *K8STestSetup) TestMainWithHook(m *testing.M, hook Hook) {
+	s.testMain(m, hook)
+}
+
+// Hook function type is invoked post-setup but pre-test
+type Hook func() error
+
+func (s *K8STestSetup) testMain(m *testing.M, hook Hook) {
```

Comment:
lets make the second param variadic so that users can pass 0 or more hooks to be run